### PR TITLE
[feat] added a default scope manager

### PR DESCRIPTION
**Pull request by fishonamos**
Modified the code so when the context is created, the Scope Manager is also initialised. This will solve #345

### CODE REVIEW

**Comment by TAdev0** [comment]
@fishonamos  hi, your code will not compile because `NewScopeManager` takes a `map[string]any` as argument. 
What you can do instead is calling `*hinter.DefaultNewScopeManager()` , and `DefaultNewScopeManager` will call `NewScopeManager` for you passing an empty array as argument.

**Comment by TAdev0** [comment]
Basically, you have 2 possibilities: 

- manually defining the ScopeManager field of `ctx` 
```
	ctx := &hinter.HintRunnerContext{
			ScopeManager: *hinter.DefaultNewScopeManager(),
		}
```
- creating an uninitialized `ctx` an calling `InitializeScopeManager` on it , passing an empty array of string
```
		ctx := &hinter.HintRunnerContext{}
		hinter.InitializeScopeManager(ctx, make(map[string]any))

```

**Comment by fishonamos** [comment]
> Basically, you have 2 possibilities:
> 
> * manually defining the ScopeManager field of `ctx`
> 
> ```
> 	ctx := &hinter.HintRunnerContext{
> 			ScopeManager: *hinter.DefaultNewScopeManager(),
> 		}
> ```
> 
> * creating an uninitialized `ctx` an calling `InitializeScopeManager` , passing `ctx`  and an empty array of string to initialize the ScopeManager field:
> 
> ```
> 		ctx := &hinter.HintRunnerContext{}
> 		hinter.InitializeScopeManager(ctx, make(map[string]any))
> ```

Thank you. Learnt a lot here. I used the Uninitialized ctx option. This creates an empty HintRunnerContext first, then  initializes the ScopeManager using InitializeScopeManager().

File: pkg/hintrunner/zero/zerohint_test.go [hunk 1]
```diff
@@ -117,9 +117,7 @@ func runHinterTests(t *testing.T, tests map[string][]hintTestCase) {
 		}
 
 		ctx := &hinter.HintRunnerContext{}
-		if tc.ctxInit != nil {
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_test.go:120]: Why did you remove these lines?

**Review by TAdev0 — CHANGES_REQUESTED** [review]

File: pkg/hintrunner/zero/zerohint_test.go (hunk 1 above)
Comment (fishonamos) [reply pkg/hintrunner/zero/zerohint_test.go:120]: Oh. A deletion error.

**Comment by TAdev0** [comment]
LGTM!

@cicr99  should we merge this like this and open another PR to modify every test to be consistent with the default initialized scope manager?

**Comment by TAdev0** [comment]
there is a unit test failing after merging the main branch.

This is becasue of `UsortVerify` test that uses `InitializeScopeManager` to create a scope with `positions_dict` . But as we already call `InitializeScopeManager` in the general test setting, it wont work.

Can you modify the `UsortVerify` so that it calls `ctx.ScopeManager.EnterScope` instead?

**Comment by TAdev0** [comment]
yes `AssignVariable` is enough, you are right. I'll do a PR to update all tests configuration once this one is merged anyway

**Comment by fishonamos** [comment]
I modified  "Usort Verify" to call just Assign Variable. I think we need to do for all others too.  Such as UsortVerifyMultiplicityAssert.

**Comment by TAdev0** [comment]
hi @fishonamos , yes i will update all other tests in another PR to make sure we never create unnecessary scopes :)

**Comment by TAdev0** [comment]
@fishonamos  your unit test dont pass. You should use `make build` and `make unit` to run tests locally and make sure they pass

**Comment by TAdev0** [comment]
you need to check the return value with `AssignVariable`

**Comment by fishonamos** [comment]
They are now passing

**Comment by TAdev0** [comment]
you need to check the return value with AssigneVariable

```

err = ctx.ScopeManager.AssignVariable("...", ...)
 if err != nil {
       return err
 }
```

**Comment by fishonamos** [comment]
All tests have passed.

**Comment by TAdev0** [comment]
lgtm!

**Review by TAdev0 — APPROVED** [review]

**Comment by TAdev0** [comment]
well we should probably have used 
```
		if err != nil {
						t.Fatal(err)
					}
```

in unit tests, but anyway it doesnt matter, and i'm pushing soon a pr that will update this

**Pull request merged**